Security Management - Improved introduction section for users, groups and roles

romartin opened the pull request:
.

mareknovotny commented:
Jenkins retest this please

mareknovotny commented on docs/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/SecurityManagement-section.adoc:
```diff
@@ -8,9 +8,20 @@ configure who can access the different resources and features available.
+
+NOTE: So keep in mind that if the security realm being used does not support both groups and roles, the user assignments will be performed with no differentiation, so all groups and roles assigned to a given user will be added in the security realm, no matter if it's a group or a role. At application's bootstrap time, it will check the Roles Registry and will apply the right considerations to dec ...
+
+NOTE: Users can be assigned with more than one role and/or group. It is always mandatory to assign at least one role to the user, otherwise login is not allowed. Roles are defined at application server level and they are part of the webapp’s deployment descriptor. On the other hand, groups are a more flexible concept, since they can be defined at runtime as well.
```
again singular is IMHO better as an entity

romartin commented:
Hey @mareknovotny . yeah agree :+1: 
Anyway I started to read the section again... and finally re-write it a little bit, I guess it results more clear now.. lemme know.
Also rebased and squashed.
Thanks!

mareknovotny commented on docs/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/SecurityManagement-section.adoc:
```diff
@@ -6,11 +6,41 @@ configure who can access the different resources and features available.
+
+* *User*
+
+A user, rather than the attributes and other kind of metadata it can define, represents the same kind entity in any of the supported security domains (Wildfly, EAP, Tomcat, Keybloak, etc), so the entity results a user in the workbench as well
```
> so the entity results a user in the workbench as well

missing preposition before  __a user__ ?

mareknovotny commented on docs/shared-kie-docs/src/main/asciidoc/Workbench/SecurityManagement/SecurityManagement-section.adoc:
```diff
@@ -6,11 +6,41 @@ configure who can access the different resources and features available.
+
+* *Role* / *Group*
+
+Both role and group are security entities, but rather than a user, the semantics, the behaviors or the structure in the domain is not usually common across environments. As an example consider that exist domains which do not support both, or domains were a the semantics for a group or a role is completely different.
```
> a the semantics
duplication of articles

romartin commented:
@mareknovotny thanks for the review! new commit added
